sync/class.jetpack-sync-defaults.php

```diff
@@ -271,6 +271,7 @@ static function is_whitelisted_option( $option ) {
 	static $default_sync_wait_threshold = 5; // only wait before next send if the current send took more than X seconds
 	static $default_max_queue_size = 1000;
 	static $default_max_queue_lag = 900; // 15 minutes
+	static $default_max_dequeue_time = 15; // 15 seconds
```

Comment:
In Slack, we discussed a couple of things, specifically 1) lowering this number so that we're not taking up half of the execution time rendering items (since we still need to send the items to WPCOM and then process the items after rendering) and 2) making this number a bit more intelligent.

To handle both of those, @lezama and I discussed something something like this.

```php
$max_exec_time = ini_get('max_execution_time');
if ( empty( $max_exec_time ) ) {
    $max_exec_time = 30;
}  
static $default_max_dequeue_time = floor( $max_exec_time / 3 );
```

Something like this should handle the case for lowering the amount of time so that we have more time to send as well as automagically adjusting based on the server configuration. What do you think @gravityrail?